### PR TITLE
completed

### PR DESCRIPTION
**Pull request by FLopez73**
# Submitting Week 1 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x ] Coding Challenges
- [x ] Coding Concepts
- [ x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -21,6 +21,12 @@ const fruit4 = "kiwi"
 // --------------------1) Create a function that takes in a number and determines if the given number is below boiling point, at boiling point, or above boiling point. Boiling point is 212 degrees Fahrenheit. Use the test variables provided below.
 
 // Pseudo code:
+//input: 1 number 
+//output: 1 string
+//create a function named boilingPoint that has a parameter of the temp for the number input
+//the function takes the parameter number and compares against the boiling point which is 212 
+//it returns a string with a inperpolation syntax to include input letting user know if above, below or at boiling.
```
Comment (Genemichael25) [inline code-challenges.js:28]: Great job on this pseudocode! I love that you mention string interpolation and how you intend to use it!! 💯

File: code-challenges.js [hunk 2]
```diff
@@ -31,20 +37,54 @@ const temperature2 = 350
...
+//using method console.log to reverse the array secondarrly running .join function to turn array back into string
+//return string in reverse order
 
 const currentCohort = "Bravo 2023"
 // Expected output: "3202 ovarB"
+ 
+//const ourCohort = (currentCohort.split(''))
+
+//console.log((ourCohort.reverse()).join(''))
+//output:3202 ovarB
```
Comment (Genemichael25) [inline code-challenges.js:86]: Nice job! Why did you use ('') in the split and join methods?

File: code-challenges.js [hunk 3]
```diff
@@ -66,3 +113,13 @@ const sanDiegoSummerTemperatures = [79, 80, 72, 73, 82, 77, 76]
 
 const sanDiegoWinterTemperatures = [59, 68, 62, 59, 66, 67, 66]
 // Expected output: [68, 67, 66, 66, 62, 59, 59]
+
+const tempOne = (sanDiegoSummerTemperatures.sort().reverse())
+
+const tempTwo = (sanDiegoWinterTemperatures.sort().reverse())
```
Comment (Genemichael25) [inline code-challenges.js:126]: Awesome! Great job getting the output you expected!! Sort allows us to pass a function that will change the way it sorts and reverse might not even be necessary if you want to check that out. Missing pseudocode for this problem.

File: coding-concepts.js [hunk 4]
```diff
@@ -7,48 +7,53 @@
...
 // --------------------4) What will this log?
 
 const weekendDays = ["saturday", "sunday"]
-// console.log(weekendDays.toUpperCase())
+//console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer:SATURDAY SUNDAY
+// b) Verify and explain:Error .toUpperCase() is a string property that is not applicable to the array since they return values.
```
Comment (Genemichael25) [inline coding-concepts.js:50]: Nice follow up! It is important to note which methods work on what data types.

File: coding-concepts.js [hunk 5]
```diff
@@ -7,48 +7,53 @@
...
 
 const dataTypes = ["number", "string", "Boolean", "undefined"]
-// console.log(typeof dataTypes.length)
+//console.log(typeof dataTypes.length)
+//output:number
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer:4
+// b) Verify and explain:number due to the "typeof" operator which checks the type of operand's value which is an array that only returns numbers.
```
Comment (Genemichael25) [inline coding-concepts.js]: typeof in this scenario is giving you the data type which results from using .length on the array

File: interview-questions.md [hunk 6]
```diff
@@ -8,50 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+https://www.w3schools.com/git/git_intro.asp?remote=github
+git-scm.com
 
 2. Which JavaScript operators will return a Boolean value?
 
-Your answer:
+Your answer: Equality, logical and relational operators will return Boolean.
 
-Researched answer:
+Researched answer:
```
Comment (Genemichael25) [inline interview-questions.md]: Missing researched answer, would be a great spot to expand on current knowledge as well as use examples.

File: code-challenges.js [hunk 7]
```diff
@@ -31,20 +37,54 @@ const temperature2 = 350
...
 const currentCohort = "Bravo 2023"
 // Expected output: "3202 ovarB"
+ 
+//const ourCohort = (currentCohort.split(''))
+
+//console.log((ourCohort.reverse()).join(''))
+//output:3202 ovarB
+
 
 // --------------------4) Create the code that will return the last index of the given value from the array using the test variables provided below.
```
Comment (Genemichael25) [inline code-challenges.js:90]: Need pseudocode for this problem. You can mention the method you intend on using and how it works.

File: code-challenges.js [hunk 8]
```diff
@@ -31,20 +37,54 @@ const temperature2 = 350
...
+    return (`${temp} below boiling point`)
+} else if (temp === 212){
+    return (`${temp} at boiling point`)
+} else {return ("who knows?")}
+}
+
+
+console.log(boilingPoint (42))  //42 above boiling point
+console.log(boilingPoint (350)) // 350 above boiling point
+console.log(boilingPoint (212)) //212 at boiling point
```
Comment (Genemichael25) [inline code-challenges.js]: You can pass the variables as arguments here.

**Review by Genemichael25 — CHANGES_REQUESTED** [review]
Felix, this is a very solid start to these assessments! I love the pseudocode you provided for # 1 on the code challenges but I am requesting changes for missing pseudocode in # 4 and 5 in code challenges. You can briefly mention the methods you intend to use and their output, or how they will work for your approach. Also, need a researched answer for # 2 in the interview questions section. Left a comment to help you expand on that, as well as a few more throughout for you to consider. Please make the changes and push up on the same branch and notify me so I can finish reviewing and approve the PR. Any questions, don't hesitate to reach out!

**Comment by FLopez73** [comment]
Hey Gene! Thank you for all the feedback it was very helpful. I don't know how I completely missed some questions lol. I have pushed the updated assessment let me know if you can see it.

File: code-challenges.js (hunk 3 above)
Comment (FLopez73) [reply code-challenges.js:126]: I get what you mean by this now after I submitted by updates. The .sort() method can take a comparison function as an argument in order to set the descending order which would be (a,b) => b-a.

**Review by Genemichael25 — APPROVED** [review]
Thanks for making those changes! Continue to use pseudocode and break down the problems to help you get to that next level. This assessment currently displays a novice level of comprehension and is a great start, keep up the good work!

**Pull request merged**